pkg/scheduler/worker/workerpool_test.go

```diff
@@ -87,3 +99,126 @@ func TestWorkerPool(t *testing.T) {
 	require.Equal(t, reconEntity.SchedulingID, testInvoker.params[0].SchedulingID)
 	require.Equal(t, opsProcessable[0].CorrelationID, testInvoker.params[0].CorrelationID)
 }
+
+func TestWorkerPoolParallel(t *testing.T) {
+
+	t.Run("Multiple WorkerPools watching same reconciliation repository", func(t *testing.T) {
+		//make sure persistent db is empty
+		err := os.RemoveAll("./test/")
+		require.NoError(t, err)
+
+		//initialize WaitGroup
+		var wg sync.WaitGroup
+		//prepare keb clusters
+		kebClusters := []*keb.Cluster{
```

Comment:
Let's reduce this big block, we can create 3 dummy cluster one by one using `pkg/cluster/inventory_test.go:302`